packages/trpc/server/routers/viewer/workflows/update.handler.ts

```diff
@@ -455,13 +453,16 @@ export const updateHandler = async ({ ctx, input }: UpdateOptions) => {
       });
 
       //cancel all workflow reminders from steps that were edited
-      remindersToUpdate.forEach(async (reminder) => {
+      const pDeleteReminders = remindersToUpdate.map((reminder) => {
         if (reminder.method === WorkflowMethods.EMAIL) {
           deleteScheduledEmailReminder(reminder.id, reminder.referenceId);
         } else if (reminder.method === WorkflowMethods.SMS) {
           deleteScheduledSMSReminder(reminder.id, reminder.referenceId);
+        } else if (reminder.method === WorkflowMethods.WHATSAPP) {
+          deleteScheduledWhatsappReminder(reminder.id, reminder.referenceId);
         }
       });
+      await Promise.all(pDeleteReminders);
```

Comment:
How do you think about wrapping the code itself in Promise.all vs storing the result in `pDeleteReminders` and then doing await.

I find the earlier approach easier to read.

Reply:
So this becomes, 
```
await Promise.all(remindersToUpdate.map((reminder) => {
        if (reminder.method === WorkflowMethods.EMAIL) {
          deleteScheduledEmailReminder(reminder.id, reminder.referenceId);
        } else if (reminder.method === WorkflowMethods.SMS) {
          deleteScheduledSMSReminder(reminder.id, reminder.referenceId);
        } else if (reminder.method === WorkflowMethods.WHATSAPP) {
          deleteScheduledWhatsappReminder(reminder.id, reminder.referenceId))
```

Reply:
Sure that works. For longer mapping functions I tend to prefer prior assignment, but this is rather short.

Reply:
Will actually revert the whole thing. One of those implicit async calls that could probably do better with `Promise.allSettled()` in another PR. Error handling should really be discussed.